### PR TITLE
Object Oriented Programming in Python

### LINKED ISSUE #2606: Object Oriented Programming with Python
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Object-Oriented Programming with Python

### Introduction paragraph (2-3 paragraphs):
Object-Oriented Programming is a technique of writing computer programs among others. It helps to modulize our code into objects that can be reused later on. This technique is supported by major popular languages including Python. 

In this tutorial, you would be exposed to object-oriented principle using Python language.

### Key takeaways: 
- Classes vs Objects
- Setters and Getters
- Magic method
- Inheritance

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Temmy5113**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #2606

### CODE REVIEW

File: content/articles/object-oriented-programming-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+### Prerequisites
+Python should be installed on your local machine. I recommend at least from version `3.7` or newer.
+
+Some knowledge of Python is also required.
+
+### Project Overview
+For this tutorial, this would be the project we'll be using to explain most of the OOP principles found in the Python programming language.
+
+```python
```
Comment (geoffrey45) [inline content/articles/object-oriented-programming-in-python/index.md:29]: Explain this using words and not code. You are introducing the reader to this concepts, so the reader doesn't understand the code.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@Temmy5113 Your article is 680 words (minus code) which is relatively shorter than the required article length (750 words). Please add more content to your article to meet the minimum requirements.

**Comment by Temmy5113** [comment]
@geoffrey45 changes have been made

File: content/articles/object-oriented-programming-in-python/index.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+excerpt_separator: <!--more-->
+images:
+    - url: /engineering-education/object-oriented-programming-in-python/hero.jpg
+      alt: OOP Python example image
+---
+
+Many modern languages like Java, C# are object-oriented or have support for using the object-oriented programming (OOP) paradigms. Python is not exempted.
+
+In this tutorial, you will learn how to use OOP concepts with Python. OOP is a technique of structuring one's code, other techniques what looking into is __functional programming__.
+
```
Comment (geoffrey45) [inline content/articles/object-oriented-programming-in-python/index.md:20]: @Temmy5113 Please explain both of these statements more clearly.

**Comment by geoffrey45** [comment]
@Temmy5113 Please create an author folder in the `/content/authors`. The folder carries your author photo and bio. Feel free to check the other student's folders.

**Comment by hectorkambow** [comment]
@geoffrey45 

Is that all this PR needs? To be ready for a final review?

**Comment by geoffrey45** [comment]
@hectorkambow Yes. Only the author folder

**Comment by Temmy5113** [comment]
@geoffrey45 made the necessary adjustment

Cc: @hectorkambow

File: content/articles/object-oriented-programming-in-python/index.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+author: joseph-abiola
+date: 2021-11-01T00:00:00-13:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/object-oriented-programming-in-python/hero.jpg
+    alt: Object Oriented Programming Python Image Hero
+---
+
+For the last four years, security researchers have reported a tremendous increase in security breaches. About 20-30,000 websites get hacked every day. There are several ways websites get hacked, XML External Entity Attacks (XXE) being one of them.
```
Comment (WanjaMIKE) [inline content/articles/object-oriented-programming-in-python/index.md:17]: @mercymeave I believe, you copied the wrong description in the front matter. Am I right?

**Comment by WanjaMIKE** [comment]
Merged #4284 into master

**Pull request merged**